### PR TITLE
[IMP] account: get default payment ref format with country

### PR DESCRIPTION
**Pull request by jbw-odoo**
Set the payment reference standard to the one of the country of the
company by default.

This is a backport of commit https://github.com/odoo/odoo/commit/1493a01b5df1339765360a059a466ecfd2a10494. Mimicking it's bahviour without
merging the l10n_be_invoice_bba module into the l10n_be module.

Task: 2440462

Description of the issue/feature this PR addresses:

### CODE REVIEW

File: addons/l10n_be_invoice_bba/i18n/l10n_be_invoice_bba.pot [hunk 1]
```diff
@@ -15,28 +15,28 @@ msgstr ""
 "Content-Transfer-Encoding: \n"
 "Plural-Forms: \n"
 
-#. module: l10n_be_invoice_bba
-#: model:ir.model.fields.selection,name:l10n_be_invoice_bba.selection__account_journal__invoice_reference_model__be
+#. module: l10n_be
```
Comment (qdp-odoo) [inline addons/l10n_be_invoice_bba/i18n/l10n_be_invoice_bba.pot:18]: mmh lloks like a file renaming is missing here?

**Comment by qdp-odoo** [comment]
hello @jbw-odoo . Runbot is red over here. Can you check please?

**Comment by jbw-odoo** [comment]
> hello @jbw-odoo . Runbot is red over here. Can you check please?

Rebased, it's green now

File: addons/account/models/account_journal.py [hunk 2]
```diff
@@ -43,6 +43,16 @@ def _get_bank_statements_available_sources(self):
...
+    
+    def _default_invoice_reference_model(self):
+        """Get the invoice reference model according to the company's country."""
+        country_code = self.env.company.country_id.code
+        country_code = country_code and country_code.lower()
+        if country_code:
+            for model in self._fields['invoice_reference_model'].get_values(self.env):
+                if model.startswith(country_code):
+                    return model
+        return 'odoo'
```
Comment (smetl) [inline addons/account/models/account_journal.py:55]:
```
@api.depends('country_code')
def _compute_invoice_reference_model(self):
    for journal in self:
        invoice_reference_model = False
        for model in self._fields['invoice_reference_model'].get_values(self.env):
            if model.startswith(country_code):
                invoice_reference_model = model
                break;
            if not invoice_reference_model:
                invoice_reference_model = 'odoo'
            journal.invoice_reference_model = invoice_reference_model

```

```
invoice_reference_model = fields.Selection(
    selection=[('odoo', 'Odoo'),('euro', 'European')],
    string='Communication Standard',
    store=True, readonly=False,
    compute='_compute_invoice_reference_model',
    help="You can choose different models for each type of reference. The default one is the Odoo reference.")
```

**Review by smetl — CHANGES_REQUESTED** [review]

File: addons/account/models/account_journal.py (hunk 2 above)
Comment (jbw-odoo) [reply addons/account/models/account_journal.py:55]: The purpose of this is to backport https://github.com/odoo/odoo/commit/1493a01b5df1339765360a059a466ecfd2a10494.
@qdp-odoo should I make this change in master too ?

Comment (smetl) [reply addons/account/models/account_journal.py:55]: @jbw-odoo I didn't understand it was a backport. I let @qdp-odoo to decide ;)

**Comment by qdp-odoo** [comment]
it's fine like this

@robodoo r+

**Pull request closed without merging**